### ISSUE TITLE
Optimize auxiliary relation

### ISSUE THREAD

**Issue by codygunton** [AztecProtocol/barretenberg#757]
`AuxiliaryRelationImpl` does not take advantage of the possibility for subrelation to have smaller degrees. This inefficiency is exacerbated when the relation is applied in the context of Protogalaxy, where the "full" degrees all receive a large adjustment. It may be possible to optimize the auxiliary relation, but it may be that the optimizations come at the expense of sharing computation between the subrelations. In any case, since the auxiliary relation is somewhat complicated, I leave it as an issue for further investigation later.

**Comment by codygunton** [comment]
Another possible optimization: using powers of $\eta$ rather than separate challenges blows up the total degrees significantly.

**Comment by ledwards2225** [comment]
@notnotraju Can you comment on whether this is still the case? I think you determined that we can't do much here

**Comment by ledwards2225** [comment]
This was evaluated by @notnotraju - nothing more to do

**Comment by notnotraju** [comment]
Yeah, basically I think the optimization right now is somewhat marginal. However, once we separate out the aux relation into the two constituent subrelations (which I suppose incurs a PG cost), as [#1448 ](https://github.com/AztecProtocol/barretenberg/issues/1448) I think this *could* maybe be nice-to-have improvement?
